src/components/data/DataNavigation.js

```diff
@@ -65,33 +64,36 @@ const useStyles = makeStyles((theme) => ({
  */
 function getPathItems(relativePath) {
     let pathItems = [];
-    pathItems = relativePath.split(PATH_SEPARATOR).slice(1);
+    pathItems = relativePath.split(constants.PATH_SEPARATOR).slice(1);
     return pathItems;
 }
 
 /**
  * Computes the path for routing.
  *  * @example
- * // returns "/data?path=/iplant/home/ipctest/analyses"
- * getPathItems("/iplant/home/ipctest","/analyses/wordcount/logs", 0);
- * @param {string }root - CyVerse Data Store root path e.g: /iplant/home/ipctest
+ * // returns "/data/ds/iplant/home/ipctest/analyses"
+ * getPathItems("ds","/iplant/home/ipctest","/analyses/wordcount/logs", 0);
+ * @param {string} storageId - storage id of data storage that is being browsed. e.g. : "ds" for
+ * CyVerse Data Store
+ * @param {string } root - CyVerse Data Store root path e.g: /iplant/home/ipctest
  * @param {string} relativePath - relative path to CyVerse Data Store e.g: /analyses/wordcount/logs
  * @param {number} selectedPathItemIndex - index of the selected path item
  * @returns {string} - path to be used by the nextjs router
  */
-function pathToRoute(root, relativePath, selectedPathItemIndex) {
+function pathToRoute(storageId, root, relativePath, selectedPathItemIndex) {
     const relativePathItems = getPathItems(relativePath);
     const reducer = (accumulator, currentVal, idx) => {
         if (idx <= selectedPathItemIndex) {
-            return accumulator + PATH_SEPARATOR + currentVal;
+            return accumulator + constants.PATH_SEPARATOR + currentVal;
         }
         return accumulator;
     };
     const routerPath = relativePathItems.reduce(reducer);
     return (
-        "/" +
+        constants.PATH_SEPARATOR +
         NavigationConstants.DATA +
-        `?path=${root}${PATH_SEPARATOR}${routerPath}`
+        constants.PATH_SEPARATOR +
+        `${storageId}${root}${constants.PATH_SEPARATOR}${routerPath}`
```

Comment:
Are there any cases where you might have to URL encode one of the path components? If so, a small function to construct URLs might be helpful.